fs/fcb/src/fcb.c

```diff
@@ -118,14 +124,29 @@ fcb_free_sector_cnt(struct fcb *fcb)
             break;
         }
     }
+
+    os_mutex_release(&fcb->f_mtx);
+
     return i;
 }
 
 int
 fcb_is_empty(struct fcb *fcb)
 {
-    return (fcb->f_active.fe_area == fcb->f_oldest &&
+    int rc = 0;
+    bool ret = false;
+
+    rc = os_mutex_pend(&fcb->f_mtx, OS_WAIT_FOREVER);
+    if (rc && rc != OS_NOT_STARTED) {
+        return FCB_ERR_ARGS;
```

Comment:
Would be good to note in header file that negative value means error and not that it is not empty (non-zero value)

(alternative would be to make this   int fcb_is_empty(struct fcb *fcb, bool *empty) for more explicit error vs return value

Reply:
I think using the return code is good enough, will just update the header to include the updated return code.